src/os_posix/os_fs.c

```diff
@@ -140,7 +140,7 @@ __posix_directory_sync(WT_SESSION_IMPL *session, const char *path)
 
 	fd = 0;				/* -Wconditional-uninitialized */
 	WT_SYSCALL_RETRY((
-	    (fd = open(dir, O_RDONLY, 0444)) == -1 ? -1 : 0), ret);
+	    (fd = open(dir, O_RDONLY | O_CLOEXEC, 0444)) == -1 ? -1 : 0), ret);
```

Comment:
Clang Tidy recommends to use this flag for calls to `open`. To my understanding, it just causes any child processes created via `fork` to not have access to any open file descriptors from the parent. I don't see a `fork` or `exec` call between here and where we call `close` so I think this change should be safe.